client/gutenberg/editor/utils.js

```diff
@@ -3,29 +3,37 @@
  * External dependencies
  */
 import apiFetch from '@wordpress/api-fetch';
-import { noop } from 'lodash';
-
-/**
- * Internal dependencies
- */
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:gutenberg' );
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import { noop, startsWith } from 'lodash';
 
 export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
-
 	const rootURL = 'https://public-api.wordpress.com/';
 	apiFetch.use( apiFetch.createRootURLMiddleware( rootURL ) );
 
-	// rewrite default API paths to match WP.com equivalents
-	// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
 	apiFetch.use( ( options, next ) => {
-		const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
-
-		debug( 'sending API request to: ', wpcomPath );
-
-		return next( { ...options, path: wpcomPath } );
+		if ( startsWith( options.path, '/wp/v2/sites/' ) ) {
+			// rewrite default API paths to match WP.com equivalents
+			// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
+			const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
+			return next( { ...options, path: wpcomPath } );
+		} else if ( startsWith( options.path, '/me' ) ) {
+			// rewrite the namespace of /me endpoints
+			// example: /wp/v2/me/settings -> /rest/v1.1/me/settings
+			return new Promise( ( resolve, reject ) => {
+				wpcomProxyRequest(
+					{
+						...options,
+						apiNamespace: 'rest/v1.1',
+					},
+					( error, body ) => {
+						if ( error ) {
+							return reject( error );
+						}
+						return resolve( body );
+					}
+				);
+			} );
+		}
+		return noop;
```

Comment:
obviously not an issue here but if we build this for realz I'm definitely thinking we need a new "project" maybe in the `~/packages` directory